pkg/services/ngalert/state/persister_async.go

```diff
@@ -0,0 +1,62 @@
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"go.opentelemetry.io/otel/trace"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+)
+
+type AsyncStatePersister struct {
+	log log.Logger
+	// doNotSaveNormalState controls whether eval.Normal state is persisted to the database and returned by get methods.
+	doNotSaveNormalState bool
+	store                InstanceStore
+}
+
+func NewAsyncStatePersisiter(log log.Logger, cfg ManagerCfg) StatePersister {
+	return &AsyncStatePersister{
+		log:                  log,
+		store:                cfg.InstanceStore,
+		doNotSaveNormalState: cfg.DoNotSaveNormalState,
+	}
+}
+
+func (a *AsyncStatePersister) Async(ctx context.Context, ticker *clock.Ticker, cache *cache) {
+infLoop:
+	for {
+		select {
+		case <-ticker.C:
+			if err := a.fullSync(ctx, cache); err != nil {
+				a.log.Error("Failed to do a full state sync to database", "err", err)
+			}
+		case <-ctx.Done():
+			a.log.Info("Stopping state sync...")
+			if err := a.fullSync(context.Background(), cache); err != nil {
+				a.log.Error("Failed to do a full state sync to database", "err", err)
+			}
+			ticker.Stop()
```

Comment:
Then this just becomes a return:

```suggestion
			return
```